### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by amejid**
## Milestone 8: Integration specs for views and fix n+1 problems

In this PR, the following were implemented:-

- Solved N + 1 problems when querying the database

Written integration tests using  Capybara for
- Users index page
- User show page
- Posts index page
- Post show page

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class PostsController < ApplicationController
   before_action :set_user, only: %i[index create show]
   def index
-    @posts = @user.posts
+    @posts = @user.posts.paginate(page: params[:page], per_page: 3)
```
Comment (Mihndim2020) [inline app/controllers/posts_controller.rb:4]: - [ ] As it is a requirement for this project, `solve N+1 problem when fetching all posts and their comments for a user`, 
Something like so: `@posts = @user.posts.includes(:comments)`, please kindly fix that.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @amejid, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (amejid) [reply app/controllers/posts_controller.rb:4]: @Mihndim2020 I am using bullet to track N + 1 problems, and this query doesn't have N+1 problem.

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED ✔🍾🎉**
Hello @amejid 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- You asked for the changes in a polite and friendly way.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by amejid** [comment]
Thanks for reviewing and approving my work.

**Pull request merged**